src/data-types/bigint.ts

```diff
@@ -15,15 +17,22 @@ const BigInt: DataType = {
     buffer.writeUInt8(8);
   },
 
-  writeParameterData: function(buffer, parameter, _options, cb) {
+  writeParameterData: function(buff, parameter, options, cb) {
+    buff.writeBuffer(Buffer.concat(Array.from(this.generate(parameter, options))));
+    cb();
+  },
+
+  generate: function* (parameter: ParameterData, options: InternalConnectionOptions) {
```

Comment:
As the parameter types for `generate` are defined in the `DataType` definition, specifying the types again here is a bit repetitive.

This could be changed to the following:

```suggestion
  generate: function* (parameter, options) {
```